sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Keys/src/AzureDataProtectionKeyVaultKeyBuilderExtensions.cs

```diff
@@ -24,9 +25,9 @@ public static class AzureDataProtectionKeyVaultKeyBuilderExtensions
         /// <param name="keyIdentifier">The Azure Key Vault key identifier used for key encryption.</param>
         /// <param name="tokenCredential">The token credential to use for authentication.</param>
         /// <returns>The value <paramref name="builder"/>.</returns>
-        public static IDataProtectionBuilder ProtectKeysWithAzureKeyVault(this IDataProtectionBuilder builder, string keyIdentifier, TokenCredential tokenCredential)
+        public static IDataProtectionBuilder ProtectKeysWithAzureKeyVault(this IDataProtectionBuilder builder, Uri keyIdentifier, TokenCredential tokenCredential)
         {
-            return ProtectKeysWithAzureKeyVault(builder, keyIdentifier, new KeyResolver(tokenCredential));
+            return ProtectKeysWithAzureKeyVault(builder, keyIdentifier.ToString(), new KeyResolver(tokenCredential));
```

Comment:
Probably should throw an `ArgumentNullException` before throwing a `NullReferenceException` here. Would be easier to diagnose.

Reply:
Good point